pkg/controller/controller.go

```diff
@@ -58,6 +58,10 @@ type Options struct {
 	// RecoverPanic indicates whether the panic caused by reconcile should be recovered.
 	// Defaults to the Controller.RecoverPanic setting from the Manager if unset.
 	RecoverPanic *bool
+
+	// LeaderElected indicates whether the controller needs to use leader election.
+	// Defaults to true, which means the controller will use leader election.
+	LeaderElected *bool
```

Comment:
IMHO the method name is clearer in terms of clarifying what this is for
```suggestion
	NeedLeaderElection *bool
```

Reply:
@alvaroaleman, I agree that the method name is clearer.

I can't use `NeedLeaderElection` for the field though since the method needs that name to implement `manager.LeaderElectionRunnable` and making the field match leads to a [DuplicateFieldAndMethod](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#DuplicateFieldAndMethod) error. Do you think any of these field names would be clearer?

* `NeedsLeaderElection` <- Just adds a `s` at the end of `Need` to prevent duplicate field
* `RequiresLeaderElection`
* `WantsLeaderElection`
* `UsesLeaderElection`

I'm open to any suggestions to make it clearer

Reply:
The field name here can be anything, this is controller.Options not controller.Controller

Reply:
I'm sorry I missed that :man_facepalming: . Thank you for pointing that out! Should be fixed in https://github.com/kubernetes-sigs/controller-runtime/pull/2117/commits/8988b10b2131766f2ea401a8687c77d7d4298453